connection.js

```diff
@@ -401,145 +436,368 @@ class Connection {
+                EndingSentence : "",                // optionnal, for example in canadian, adds explanation
+                ConnBelowAboveSentence : "",        // for conn_log : below/above
+                ConnSentence : "",                  // for conn_log sentence
+                TimecontrolString : "",             /*"fischer" , "simple", "byoyomi" , "canadian" , "absolute"*/
+                SpeedString : "",                   /* "blitz" , "live" , "corr" */
+            };
+
+            // before starting, general information : 
+            // 0) "none" dont have a period time, so it is rejected before the UHMAEATs
+            // (we have sanity checks just in case)
+            // 1) simple time doesn't have a main time, only a period time, 
+            // so we let it slide from maintime rejects
+            // 2) fischer : doesnt have a minperiods or maxperiods
+            // 3) absolute doesnt have a period time
+            // so we let it slide from periodtime rejects
+            // 4) while using gedit "find and replace", make sure you dont replace
+            // t.max_time to t.min_time ! (it doesnt exist !)
+
+            // before starting, make sures : we refresh values //
+            // now just before TimecontrolString is being tested, we again make sure it has the latest value
+            /*"fischer", "byoyomi", "canadian", "simple", "absolute" */
+            this.TimecontrolString = String(t.time_control);
+            // now just before SpeedString is being tested, we again make sure it has the latest value
+            /* "blitz", "live", "correspondence" */
+            this.SpeedString = String(t.speed);
+
+            // before starting, sanity checks //
+            // sanity check : if not fischer, not byoyomi, not canadian, not simple, not absolute
+            if ((this.TimecontrolString !== "fischer") && (this.TimecontrolString !== "byoyomi") && (this.TimecontrolString !== "canadian") && (this.TimecontrolString !== "simple") && (this.TimecontrolString !== "absolute")) {
+                conn_log ("error, could not find allowed time control in " + t.time_control);
+                return { reject : true, msg: "error, could not find allowed time control in " + t.timecontrol};
+            }
+            // sanity check : if not "blitz" , not "live" , not "correspondence"
+            if ((this.SpeedString !== "blitz") && (this.SpeedString !== "live") && (this.SpeedString !== "correspondence")) {
+                conn_log ("error, could not find allowed game speed in " + t.speed);
+                return { reject : true, msg: "error, could not find allowed game speed in " + t.speed};
+            }
+            // -> then if sanity checks all pass :       
+
+            //////////// for blitz games : "blitz" //////////////////
+            // min
+            if ((config.minmaintimeblitz || config.minmaintimeblitzranked || config.minmaintimeblitzunranked) && (this.SpeedString === "blitz")) {
```

Comment:
For sanity, please put the `(this.SpeedString === "blitz")` check first. Makes it more readable to me.

Actually we already check the seperate settings down here, it might be enough to just check fror speed=blitz here. Simplifies the code and the performance is really neglectable.

Reply:
actually i did it for the exact same reason : for performance

i thought if this check fails then we dont waste cpu power to check all the checks below !

and for blitz yes it is clearer, similar to what i did in #121